docs(examples): add SAM example for large-messages module

sasanchezramirez opened the pull request:
## Summary

This PR introduces a comprehensive SAM example for the `powertools-large-messages` module. It demonstrates how to use the functional interface approach (`LargeMessages.processLargeMessage`) to handle SQS messages with offloaded S3 payloads, following the request in the linked issue.

### Changes

- Created new module `powertools-examples-large-messages` under `examples/`.
- Added `template.yaml` configuring S3, SQS, and Lambda with correct IAM permissions (S3CrudPolicy).
- Implemented `App.java` demonstrating `LargeMessages.processLargeMessage()` usage (functional API).
- Configured `pom.xml` with the latest SQS Extended Client dependency.
- Added `README.md` with deployment and testing instructions.
- Added `log4j2.xml` resource for proper logging output.

**Issue number:** Closes #1353

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the code style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Changes generate no new warnings

phipag commented:
@sasanchezramirez Can you please update the description of this PR to match our PR template here? https://github.com/aws-powertools/powertools-lambda-java/blob/main/.github/PULL_REQUEST_TEMPLATE.md

Especially, make sure that the disclaimer is included (bottom section).

phipag commented on examples/powertools-examples-large-messages/README.md:
```diff
@@ -0,0 +1,27 @@
+# Powertools for AWS Lambda (Java) - Large Messages Example
+
+This project contains an example of a Lambda function using the **Large Messages** module of Powertools for AWS Lambda (Java). For more information on this module, please refer to the [documentation](https://docs.powertools.aws.dev/lambda-java/utilities/large_messages/).
```
```suggestion
This project contains an example of a Lambda function using the **Large Messages** module of Powertools for AWS Lambda (Java). For more information on this module, please refer to the [documentation](https://docs.aws.amazon.com/powertools/java/latest/utilities/large_messages/).
```

phipag commented on examples/powertools-examples-large-messages/src/main/java/helloworld/App.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public final class App implements RequestHandler<SQSEvent, String> {
+
+    private static final Logger LOG = LogManager.getLogger(App.class);
```
Following our documentation, let's use SLF4J here instead of direct calls to Log4j2. Also, we forgot to add the `@Logging` annotation to the `handleRequest` method.

https://docs.aws.amazon.com/powertools/java/latest/core/logging/

phipag reviewed (changes requested):
Hey @sasanchezramirez,

thanks for sending this PR. It is a great start! Beyond the comments below, I would like to provide a full end-to-end example here. Currently, we are missing the producer side using the SQS extended lib client.

My suggestion is this:
- Create a new `tools/` folder
- In this folder provide a small tool that can be executed locally using the maven exec plugin
- This should contain a small Java script that uses the SQS extended client to send a large payload loaded from e.g. a JSON file to the Lambda. 
- Add instructions on how to run this producer tool to the README.md

This will show users not only how to consume messages but gives them a starting point with a consumer **and** a producer function.

Note: We have such a producer tool already for our Kafka example here: https://github.com/aws-powertools/powertools-lambda-java/tree/main/examples/powertools-examples-kafka/tools

phipag commented:
Hey @sasanchezramirez,

let me know if you still would like to work on this. Otherwise, I can also take over.

sasanchezramirez commented:
Hi @phipag,

I've pushed the requested changes:

- Added tools folder with the SQS producer (using SQS Extended Client).
- Updated App.java to use SLF4J and @Logging.
- Updated README.md with producer instructions.
- Removed unused maven properties/dependencies.

Thank you very much for the feedback and for the opportunity to contribute to this project, ready for review!

phipag commented on examples/powertools-examples-large-messages/pom.xml:
```diff
@@ -0,0 +1,100 @@
+
+    <groupId>software.amazon.lambda.examples</groupId>
+    <artifactId>powertools-examples-large-messages</artifactId>
+    <version>2.8.0</version>
```
Can you update to version 2.9.0 which we released in the meantime?
